packages/graph-explorer/src/connector/useGEFetch.ts

```diff
@@ -43,6 +43,9 @@ const useGEFetch = () => {
     async (url, options) => {
       const response = await fetch(url, options);
       const data = await response.json();
+      if (typeof options.successCallback === "function") {
+        options.successCallback();
+      }
```

Comment:
This will miss a couple of scenarios:

- We get a response, but the JSON decoding fails
- The connection is dropped during the `fetch`

Reply:
One possible solution is to call the callback in all scenarios to ensure the queryId is cleared.

TanStack Query has a similar concept, and they named the callback `onSettled`, which I think is fitting.

https://tanstack.com/query/latest/docs/framework/react/reference/useMutation

Reply:
replaced with try/finally